packages/@okta/vuepress-site/docs/reference/api/schemas/index.md

```diff
@@ -2462,3 +2464,353 @@ Specific property types support a subset of [JSON Schema validations](https://to
+                    "type": "string",
+                    "required": true,
+                    "minLength": 5,
+                    "maxLength": 100,
+                    "permissions": [
+                        {
+                            "principal": "SELF",
+                            "action": "READ_WRITE"
+                        }
+                    ]
+                },
+                "firstName": {
+                    "title": "First name",
+                    "type": "string",
+                    "required": true,
+                    "minLength": 1,
+                    "maxLength": 50,
+                    "permissions": [
+                        {
+                            "principal": "SELF",
+                            "action": "READ_WRITE"
+                        }
+                    ]
+                },
+                "lastName": {
+                    "title": "Last name",
+                    "type": "string",
+                    "required": true,
+                    "minLength": 1,
+                    "maxLength": 50,
+                    "permissions": [
+                        {
+                            "principal": "SELF",
+                            "action": "READ_WRITE"
+                        }
+                    ]
+                },
+                "email": {
+                    "title": "Primary email",
+                    "type": "string",
+                    "required": true,
+                    "format": "email",
+                    "permissions": [
+                        {
+                            "principal": "SELF",
+                            "action": "READ_WRITE"
+                        }
+                    ]
+                }
+            },
+            "required": [
+                "login",
+                "firstName",
+                "lastName",
+                "email"
+            ]
+        },
+        "custom": {
+            "id": "#custom",
+            "type": "object",
+            "properties": {
+              "twitterUserName": {
+                  "title": "Twitter username",
+                  "description": "User's username for twitter.com",
+                  "type": "string",
+                  "required": false,
+                  "unique": "PENDING_UNIQUENESS",
+                  "minLength": 1,
+                  "maxLength": 10,
+                  "permissions": [
+                      {
+                          "principal": "SELF",
+                          "action": "READ_ONLY"
+                      }
+                  ]
+              }
+            },
+            "required": []
+        }
+    },
+    "type": "object",
+    "properties": {
+        "profile": {
+            "allOf": [
+                {
+                    "$ref": "#/definitions/base"
+                },
+                {
+                    "$ref": "#/definitions/custom"
+                }
+            ]
+        }
+    }
+}
+```
```

Comment:
One big gap: You never show how to change a unique property to be non-unique.

Reply:
Good point.  This is doubly worth calling out because the reverse operation (marking unique) is asynchronous, whereas I don't believe removing uniqueness is.

Reply:
Removing uniqueness is also "partially" asynchronous. The property becomes not unique immediately but a job is scheduled to cleanup the DB table. So it appears synchronous to the users.

Reply:
Hm, I don't see the schema id in the `UniqueEntityProperty` table, just schema type (so that one user and one group can each have the same value for a like-named unique property).  So when property is changed from unique to non-unique in user type A, how do we manage to erase only those entries associated with users of that type?  (Looking... yes, I see the job does pass the type id into the scrubber method, and we do a Join with the user table to identify records for users of that type.)

So I was able to answer my own question there, but I still have this question: If a property is changed from unique to non-unique (scrubber job is scheduled) and then changed back to unique (validation job is scheduled), how do we handle the race between those two jobs?